src/os/bluestore/BlueStore.h

```diff
@@ -1148,6 +1148,16 @@ class BlueStore : public ObjectStore,
     void rewrite_omap_key(const std::string& old, std::string *out);
     void get_omap_tail(std::string *out);
     void decode_omap_key(const std::string& key, std::string *user_key);
+
+    // Return the offset of an object on disk.  This function is intended *only*
+    // for use with zoned storage devices because in these devices, the objects
+    // are laid out contiguously on disk, which is not the case in general.
+    // Also, it should always be called after calling extent_map.fault_range(),
+    // so that the extent map is loaded.
+    int64_t get_ondisk_offset() const {
```

Comment:
May be rename to get_ondisk_starting_offset? To better reflect the purpose of the func..
Or pass logical_offset parameter to make this func general-purpose?

Reply:
Renamed it for now.  If need be, will generalize it.